### ISSUE TITLE
Error: Can't set headers after they are sent.

### ISSUE THREAD

**Issue by cappster** [microcipcip/cookie-universal#4]
Hello.
In SSR middleware i sent request in promise. In catch block i called `app.$cookie.remove` and got `Error: Can't set headers after they are sent.` I know, that in this moment i try to edit response header. But how i must use remove function in this situation?

**Comment by microcipcip** [comment]
The error seems to suggest that the headers have already been sent therefore you can't do anything about it, however the remove function works only on the request, not the response. It can't remove from the response object, because you (should) have control over it and not set the cookie in the first place.

**Comment by SebastianMieszczanczyk** [comment]
Hello. I have a question about this. Nuxt is SSR so some of the code can be run on the server side. I see that it is possible to have access to the cookie by `req`. So the questions are:
1. Don't you think that the library should support this? In other cases I don't see any arguments to have this library.

**Comment by microcipcip** [comment]
Hi @SebastianMieszczanczyk, this library already allows you to access `req` from Nuxt Middleware, have a look at the examples.

**Comment by SebastianMieszczanczyk** [comment]
I see that there is an option:
```
// nuxt middleware
export default ({ app }) => {
  const cookieRes = app.$cookies.get('cookie-name')
  const cookieRes = app.$cookies.get('cookie-name', { fromRes: true }) // get from res instead of req
  // returns the cookie value or undefined
}

// client
const cookieRes = this.$cookies.get('cookie-name')
// returns the cookie value or undefined
```

But in that case, the option is to get cookie from `res` instead of `req`. So I think it is the default behavior that it takes from `req` if it is server-side. In that way, you have a bug which is causing error message as in the topic.

**Comment by microcipcip** [comment]
@SebastianMieszczanczyk Yes it does read from `req` by default. I fail to understand what is the requested change though.

**Comment by microcipcip** [comment]
Are you implying that I should allow set/remove in the `res` object? I tried implementing this but it added a lot of extra complexity to the library, so I decided to keep it simple. 

At the end of the day, if you have control of the node middleware, you should be able to not set the cookie at all, instead of setting it and then attempting to delete it from `res`. Basically if I want with this lib I can easily delete a cookie from `req`, but I can't set it and immediately try to delete it from `res`, which IMHO is an error in the app architecture if you are trying to do it. Hope my logic makes sense.

**Comment by SebastianMieszczanczyk** [comment]
@microcipcip 
Sorry but I don't understand. Let me give you an example:

I set the auth cookie inside store action when the app is initialized. So there is a possibility that this will fire on the server-side (which I believe is causing this issue). So what you are saying is that I am not able to set this cookie when code is executing on the server-side(when I am making authorization)? As far as I understand - I can read cookies on ss from req/res but not set it right?

**Comment by microcipcip** [comment]
@SebastianMieszczanczyk That is not correct. You can `get` and `set` a cookie Server Side, what you cannot do is delete it server side immediately after it has been set.

For instance, all these examples are for SSR:
```
// This is possible
app.$cookies.get('cookie-name')

// This is also possible
app.$cookies.set('cookie-name')
app.$cookies.get('cookie-name', { fromRes: true } // I read the cookie immediately (but from RES)

// This is possible
app.$cookies.remove('cookie-name')

// This is not possible
app.$cookies.set('cookie-name')
app.$cookies.remove('cookie-name') // This cannot be delete as we cannot delete from RES with this lib
```
